app/models/manageiq/providers/ansible_tower/shared/inventory/parser/automation_manager.rb

```diff
@@ -122,7 +122,7 @@ def credentials
     collector.credentials.each do |credential|
       inventory_object = persister.credentials.find_or_build(credential.id.to_s)
       inventory_object.name = credential.name
-      inventory_object.userid = credential.username
+      inventory_object.userid = credential.username if credential.respond_to?(:username)
```

Comment:
You can do `inventory_object.userid = credential.try(:username)`

And for the UI issue that you mention, let us know what exactly that is and it's likely better to be done in UI side.

Reply:
I'm just theorizing that the UI may barf if .userid is null. (It has in a totally unrelated case I ran into last week).

Reply:
IMO if it fails we should fix it given that this is a valid case for this provider, @jameswnl can you test this on the ansible provider screens?  I tested a nil userid with vmware credentials and it didn't blow up.

Reply:
I tested external Tower provider case and here're what I found:

1.  There's no CFME UI showing the credentials
2. The `userid`  is "" (empty string) for a credential that has no `username` defined.

So the following would work 
`inventory_object.userid = credential.try(:username) || ""`

(I didn't hit the exception because of the issue in Ansible Tower gem)